src/main/java/com/mongodb/hibernate/jdbc/ConnectionAdapter.java

```diff
@@ -47,275 +47,275 @@ abstract class ConnectionAdapter implements Connection {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("prepareStatement not implemented");
     }
 
     @Override
     public CallableStatement prepareCall(
             String sql, int resultSetType, int resultSetConcurrency, int resultSetHoldability) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("prepareCall not implemented");
     }
 
     @Override
     public PreparedStatement prepareStatement(String sql, int autoGeneratedKeys) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("prepareStatement not implemented");
     }
 
     @Override
     public PreparedStatement prepareStatement(String sql, int[] columnIndexes) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("prepareStatement not implemented");
     }
 
     @Override
     public PreparedStatement prepareStatement(String sql, String[] columnNames) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("prepareStatement not implemented");
     }
 
     @Override
     public Clob createClob() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("createClob not implemented");
     }
 
     @Override
     public Blob createBlob() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("createBlob not implemented");
     }
 
     @Override
     public NClob createNClob() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("createNClob not implemented");
     }
 
     @Override
     public SQLXML createSQLXML() throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("createSQLXML not implemented");
     }
 
     @Override
     public boolean isValid(int timeout) throws SQLException {
-        throw new SQLFeatureNotSupportedException();
+        throw new SQLFeatureNotSupportedException("isValid not implemented");
     }
 
     @Override
     public void setClientInfo(String name, String value) throws SQLClientInfoException {
-        throw new SQLClientInfoException("won't be used", Collections.emptyMap());
+        throw new SQLClientInfoException("setClientInfo not implemented", Collections.emptyMap());
     }
 
     @Override
     public void setClientInfo(Properties properties) throws SQLClientInfoException {
-        throw new SQLClientInfoException("won't be used", Collections.emptyMap());
+        throw new SQLClientInfoException("setClientInfo not implemented", Collections.emptyMap());
     }
```

Comment:
I think, I missed this previously. Why is `SQLClientInfoException` thrown here? It seems to be that we should throw `SQLFeatureNotSupportedException`.

Reply:
Because the `SQLClientInfoException` is dictated by JDBC for the method signature, not `SQLException`. So `SQLFeatureNotSupportedException` won't make compiler happy for it is not a child class of the checked exception type in method signature.

Reply:
Apologies, I didn't notice that the `throws` clause on these methods is different.